### DISCUSSION TITLE
Disable Diagnostics?

### DISCUSSION THREAD

**Discussion by zach-is-my-name** [VonHeikemen/lsp-zero.nvim#13] [Q&A] [answered]
Is there a way I can disable diagnostics?  

Background: First time using lsp.  Project is javascript/React.  Despite having a `.jsconfig.json` in project root specifying "javascript only" I'm receiving (unwanted) typescript diagnostic messages.  (Logical follow-up question: which part of lsp-zero docs refers to configuring the underlying servers?)  

Thanks a bunch

**Comment by VonHeikemen** [comment, marked as answer]
To disable all diagnostic you can use this command:

```vim
:lua vim.diagnostic.disable()
```

The way you pass options to a server with `lsp-zero` is using the `.configure()` function. Example.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

lsp.configure('tsserver', {
  flags = {
    debounce_text_changes = 150,
  },
})

lsp.setup()
```

The first argument to `.configure()` is the name of the server and the second are the options. Since `lsp-zero` uses `lspconfig` to configure the servers the details about available options can be found in `:help lspconfig-setup`.

> and if you want to go even further, `lspconfig` uses the built-in function `vim.lsp.start_client()`. See `:help vim.lsp.start_client()`.

But I get the feeling that the configuration you want to tweak should be done in `jsconfig.json`.
